handlers/jiochat/jiochat.go

```diff
@@ -41,10 +40,15 @@ func init() {
 
 type handler struct {
 	handlers.BaseHandler
+
+	fetchTokenMutex sync.Mutex
```

Comment:
synchronizes all jiochat token lookups but we don't do a lot of traffic over jiochat and it's almost always a single redis GET so not too worried